src/commands/lambda/__tests__/uninstrument.test.ts

```diff
@@ -567,7 +567,7 @@ describe('lambda', () => {
         expect(output).toMatchInlineSnapshot(`
           "
           🐶 Uninstrumenting Lambda function
-          [Error] Couldn't fetch Lambda functions. Error: Max retry count exceeded. Error: ListFunctionsError
+          [Error] Couldn't fetch Lambda functions. Error: ListFunctionsError
```

Comment:
Would it be possible to add the reason why it couldn't fetch the lambda functions? Like a 403, 500, etc?

Reply:
I guess if we have multiple retries and they fail for different reasons it would be a bit difficult, so I'm not sure if this is worth the effort, but it would be nice to have.

Reply:
The error should bubble up from the last request that caused a failure. Here's what was shared from a customer that prompted this change:

![Image 2023-11-16 at 4 39 20 PM](https://github.com/DataDog/datadog-ci/assets/35278470/bbe433c4-1842-43f6-a106-3d77078ea9a8)

`ListFunctionsError` is just a generic error for the purposes of unit testing.

Let me know if you think additional changes are needed though!